game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -1,35 +1,50 @@
 package io.aigar.game
 
-import com.github.jpbetz.subspace.Vector2
-
-object Resources {
-  final val MaxRegular = 200
-  final val MaxSilver = 100
-  final val MaxGold = 50
+object Regular {
+  final val Max = 250
+  final val Min = 100
+  final val Mass = 1
+  final val Score = 0
 }
 
-class Resources(grid: Grid) {
-  var regular = initRegular
-  var silver = initSilver
-  var gold = initGold
+object Silver {
+  final val Max = 100
+  final val Min = 50
+  final val Mass = 3
+  final val Score = 0
+}
 
-  def initRegular: List[Vector2] ={
-    List.fill(Resources.MaxRegular)(grid.randomPosition)
-  }
+object Gold {
+  final val Max = 50
+  final val Min = 25
+  final val Mass = 0
+  final val Score = 3
+}
 
-  def initSilver: List[Vector2] ={
-    List.fill(Resources.MaxSilver)(grid.randomPosition)
-  }
+class Resources(grid: Grid) {
+  val regular = new ResourceType(grid, Regular.Min, Regular.Max, Regular.Mass, Regular.Score)
+  val silver = new ResourceType(grid, Silver.Min, Silver.Max, Silver.Mass, Silver.Score)
+  val gold = new ResourceType(grid, Gold.Min, Gold.Max, Gold.Mass, Gold.Score)
+  var resourceTypes = List(regular, silver, gold)
 
-  def initGold: List[Vector2] ={
-    List.fill(Resources.MaxGold)(grid.randomPosition)
+  def update: Unit = {
+    resourceTypes.foreach(_.spawnResources)
   }
 
   def state = {
     serializable.Resources(
-      regular,
-      silver,
-      gold
+      regular.positions,
+      silver.positions,
+      gold.positions
     )
   }
 }
+
+class ResourceType(grid:Grid, val min: Int, val max: Int, mass: Int, score: Int) {
```

Comment:
I proposed that too, but Mic brought up that it *does* create the confusion between a single `resource` in the map and the resource type. :/